go-controller/pkg/ovn/pods.go

```diff
@@ -331,9 +330,26 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 
 	logrus.Debugf("Annotation values: ip=%s ; mac=%s ; gw=%s\nAnnotation=%s",
 		podCIDR, podMac, gatewayIP, marshalledAnnotation)
-	if err = oc.kube.SetAnnotationOnPod(pod, "ovn", marshalledAnnotation); err != nil {
-		return fmt.Errorf("Failed to set annotation on pod %s - %v", pod.Name, err)
+
+	// Need to set pod annotation using both legacy and new key. This is in case if there
```

Comment:
I did some profiling and found that setting the annotation can cause, easily, 30% of the pod creation latency. Can you refactor this to set multiple annotations in a single API call?

Reply:
@girishmg 
```
diff --git a/go-controller/pkg/kube/kube.go b/go-controller/pkg/kube/kube.go
index c15fd8fdd4b4a..7f58f952ab781 100644
--- a/go-controller/pkg/kube/kube.go
+++ b/go-controller/pkg/kube/kube.go
@@ -19,6 +19,7 @@ import (
 // kubernetes resources
 type Interface interface {
 	SetAnnotationOnPod(pod *kapi.Pod, key, value string) error
+	SetAnnotationsOnPod(pod *kapi.Pod, annotations map[string]string) error
 	SetAnnotationOnNode(node *kapi.Node, key, value string) error
 	SetAnnotationsOnNode(node *kapi.Node, annotations map[string]string) error
 	UpdateNodeStatus(node *kapi.Node) error
@@ -55,6 +56,33 @@ func (k *Kube) SetAnnotationOnPod(pod *kapi.Pod, key, value string) error {
 	return err
 }
 
+// SetAnnotationsOnPod takes the pod object and map of key/value string pairs to set as annotations
+func (k *Kube) SetAnnotationsOnPod(pod *kapi.Pod, annotations map[string]string) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"annotations": annotations,
+		},
+	}
+
+	podDesc := pod.Namespace + "/" + pod.Name
+	logrus.Infof("Setting annotations %v on pod %s", annotations, podDesc)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		logrus.Errorf("Error in setting annotations on pod %s: %v", podDesc, err)
+		return err
+	}
+
+	_, err = k.KClient.CoreV1().Pods(pod.Namespace).Patch(pod.Name, types.MergePatchType, patchData)
+	if err != nil {
+		logrus.Errorf("Error in setting annotation on pod %s: %v", podDesc, err)
+	}
+	return err
+}
+
 // SetAnnotationOnNode takes the node object and key/value string pair to set it as an annotation
 func (k *Kube) SetAnnotationOnNode(node *kapi.Node, key, value string) error {
 	logrus.Infof("Setting annotations %s=%s on node %s", key, value, node.Name)
```

Reply:
@dcbw @squeed Will do. Thanks.